client/server/pages/index.js

```diff
@@ -331,10 +331,12 @@ function setUpLoggedInRoute( req, res, next ) {
 
 				if ( data.localeSlug ) {
 					req.context.lang = data.localeSlug;
-					req.context.store.dispatch( {
-						type: LOCALE_SET,
-						localeSlug: data.localeSlug,
-						localeVariant: data.localeVariant,
+					import( 'calypso/state/ui/init' ).then( () => {
+						req.context.store.dispatch( {
```

Comment:
Wanna refactor this to use the `setLocale`  action creator?

Reply:
Refactoring this might introduce timing issues; importing that action creator would likely result in attempting to initialise `ui` before a store is created, and subtly change timings on the server. I have to end up reverting the change here anyway, because it was breaking the server unit tests 😕 

`ui` and other "core" portions of state are where we're going to start seeing more of these timing issues.

Reply:
🤔  but in any case, this means that we might be dispatching this action before the store has initialized will still be a problem, no?

Reply:
Not quite; what we have here happens later. Namely, it happens after the user has been bootstrapped.

To explain further: if we were to use the action creator, it means we would have to import it. We either import it statically, or dynamically.

If we were to import it dynamically, we would have the same issue as the breaking tests in the reverted code.

If we were to import it statically, however, that would mean the imported module would be executed at *module-loading* time, rather than when this method runs. Since the imported module imports `init`, that means that `init` would now execute before anything in `client/server/pages/index.js`, as opposed to after the user has been bootstrapped.

Reply:
I understand, thanks for elaborating 👍